src/ZfcRbac/Factory/RouteGuardFactory.php

```diff
@@ -39,28 +38,40 @@ class RouteGuardFactory implements FactoryInterface, MutableCreationOptionsInter
     /**
      * {@inheritDoc}
      */
-    public function setCreationOptions(array $options)
+    public function __construct(array $options = [])
     {
         $this->options = $options;
     }
 
     /**
-     * {@inheritDoc}
+     * @param ContainerInterface $container
      * @return RouteGuard
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container)
     {
-        $parentLocator = $serviceLocator->getServiceLocator();
+        if (! method_exists($container, 'build')) { // servicemanager v3
```

Comment:
this would be not required if `$container->getServiceLocator()` moved to `createService()`